control-plane-operator/controllers/hostedcontrolplane/routecm/deployment.go

```diff
@@ -100,6 +102,13 @@ func buildRouteOCMContainerMain(image string) func(*corev1.Container) {
 			"--config",
 			path.Join(volumeMounts.Path(c.Name, routeOCMVolumeConfig().Name), configKey),
 		}
+		// For 4.14 onwards route-controller-manager uses library-go way of passing the kubeconfig via the flag
+		if ocpVersion.Minor >= 14 {
```

Comment:
the control plane operator is tied to the OCP release payload.
If the HC running is ocpVersion >= 14 then the underlying CPO is coming from the same verision. Therefore no `if ocpVersion.Minor >= 14 {` check is needed. If flags needs to be changed you just add them here and as a consumer updates the HC.spec.release, the CPO for that release will be rolled out and it will run whatever flags are in the CPO for that verion

Reply:
I see, it uses the same version as CPO. Removed the version checks